crates/fuel-gas-price-algorithm/gas-price-analysis/src/main.rs

```diff
@@ -70,83 +58,141 @@ enum Mode {
-            let (results, (p, d)) = naive_optimisation(iterations as usize, size);
+        }
+        Mode::Optimization { iterations, source } => {
+            let da_cost_per_byte =
+                get_da_cost_per_byte_from_source(source, UPDATE_PERIOD);
+            let size = da_cost_per_byte.len();
+            println!(
+                "Running optimization with {iterations} iterations and {size} blocks"
+            );
+            let simulator = Simulator::new(da_cost_per_byte);
+            let (results, (p, d)) =
+                naive_optimisation(simulator, iterations as usize, UPDATE_PERIOD);
             println!("Optimization results: P: {}, D: {}", pretty(p), pretty(d));
             (results, (p, d))
         }
     };
 
+    print_info(&results);
+
     if let Some(file_path) = &args.file_path {
         draw_chart(results, p_comp, d_comp, file_path);
-
     }
 }
 
-fn draw_chart(results: SimulationResults, p_comp: i64, d_comp: i64, file_path: &str) {
+fn print_info(results: &SimulationResults) {
     let SimulationResults {
-        gas_prices,
-        exec_gas_prices,
         da_gas_prices,
-        fullness,
-        bytes_and_costs,
         actual_profit,
         projected_profit,
-        pessimistic_costs,
+        ..
     } = results;
 
-    let max_actual_profit = pretty(*actual_profit.iter().max().unwrap() as u64);
-    println!("max_actual: {max_actual_profit}");
-
-    let plot_width = 640 * 2 * 2;
-    let plot_height = 480 * 3;
-
-    let root =
-        BitMapBackend::new(file_path, (plot_width, plot_height)).into_drawing_area();
-    root.fill(&WHITE).unwrap();
-    let (window_one, lower) = root.split_vertically(plot_height / 4);
-    let (window_two, new_lower) = lower.split_vertically(plot_height / 4);
-    let (window_three, window_four) = new_lower.split_vertically(plot_height / 4);
-
-    draw_fullness(&window_one, &fullness, "Fullness");
-
-    draw_bytes_and_cost_per_block(&window_two, &bytes_and_costs, "Bytes Per Block");
-
-    draw_profit(
-        &window_three,
-        &actual_profit,
-        &projected_profit,
-        &pessimistic_costs,
-        &format!("Profit p_comp: {p_comp:?}, d_comp: {d_comp:?}"),
-    );
-    draw_gas_prices(
-        &window_four,
-        &gas_prices,
-        &exec_gas_prices,
-        &da_gas_prices,
-        "Gas Prices",
+    // Max actual profit
+    let (index, max_actual_profit) = actual_profit
+        .iter()
+        .enumerate()
+        .max_by(|(_, a), (_, b)| a.cmp(b))
+        .unwrap();
+    let eth = *max_actual_profit as f64 / (10_f64).powf(18.);
+    println!("max actual profit: {} ETH at {}", eth, index);
+
+    // Max projected profit
+    let (index, max_projected_profit) = projected_profit
+        .iter()
+        .enumerate()
+        .max_by(|(_, a), (_, b)| a.cmp(b))
+        .unwrap();
+    let eth = *max_projected_profit as f64 / (10_f64).powf(18.);
```

Comment:
nit: I see this used in multiple places. worth an inline fn?
```suggestion
#[inline]
fn fuel_to_eth(fuel_units: i128) -> f64 {
  (fuel_units as f64) / (10_f64).powf(18.)
}
...
    let eth = fuel_to_eth(*max_projected_profit);
```

Reply:
It's a good idea, but not all of the types are `i128`. And `Into<f64>` doesn't exist for `i128`. Have to rely on the cast for now. Will get better when I clean up these types in the followup issue.

(Unless there is a way to constrain by casts?)

Reply:
yup, idt there is a way to constrain by casts. resolving.